aws-s3-transfer-manager/src/operation/upload/handle.rs

```diff
@@ -205,16 +205,29 @@ async fn complete_upload(handle: UploadHandle) -> Result<UploadOutput, crate::er
                         .set_parts(Some(all_parts))
                         .build(),
                 )
-                // TODO(aws-sdk-rust#1159) - implement checksums
-                // .set_checksum_crc32()
-                // .set_checksum_crc32_c()
-                // .set_checksum_sha1()
-                // .set_checksum_sha256()
                 .set_request_payer(handle.ctx.request.request_payer.clone())
                 .set_expected_bucket_owner(handle.ctx.request.expected_bucket_owner.clone())
                 .set_sse_customer_algorithm(handle.ctx.request.sse_customer_algorithm.clone())
                 .set_sse_customer_key(handle.ctx.request.sse_customer_key.clone())
-                .set_sse_customer_key_md5(handle.ctx.request.sse_customer_key_md5.clone())
+                .set_sse_customer_key_md5(handle.ctx.request.sse_customer_key_md5.clone());
+
+            if let Some(checksum_strategy) = &handle.ctx.request.checksum_strategy {
+                // TODO(aws-s3-transfer-manager-rs#3): allow user to pass full-object checksum value via callback on PartStream
+
+                if let Some(value) = &checksum_strategy.full_object_checksum {
+                    // We have the full-object checksum value, so set it
+                    req = match &checksum_strategy.algorithm {
+                        aws_sdk_s3::types::ChecksumAlgorithm::Crc32 => req.checksum_crc32(value),
+                        aws_sdk_s3::types::ChecksumAlgorithm::Crc32C => req.checksum_crc32_c(value),
+                        aws_sdk_s3::types::ChecksumAlgorithm::Crc64Nvme => {
+                            req.checksum_crc64_nvme(value)
+                        }
+                        algo => panic!("unexpected algorithm `{algo}` for full object checksum"),
```

Comment:
if this is meant to be a state a user can't get into at this point you might use `unreachable!` instead

Reply:
Oh cool, didn't know about `unreachable!`